### ISSUE TITLE
Windows Support (forward slashes instead of backslashes)

### ISSUE THREAD

**Issue by FaKeller** [sverweij/dependency-cruiser#27]
On Windows, dependency-cruiser extracts all modules and dependencies with backslashes `\` in their names instead of forward slashes `/`. The backslashes are even shown in the HTML tables generated by the HTML reporter.

The main problem is that the default rules don't match backslashes as they are written with forward slashes. Thus, the rules cannot be written platform-agnostic.

### Expected Behavior
dependency-cruiser should always use forward slashes.

### Possible Solution
Maybe convert all backslashes to forward slashes?

### Steps to Reproduce (for bugs)

On windows:
```sh
$ depcruise --validate .dependency-cruiser.json src

  error some-rule: src\cli.ts → src\content\entry\index.ts
...
```

### Context
Development on Windows, CI/CD on Linux. I'd love to use dependency-cruiser without any changes on both platforms.

### Your Environment

* Version used: 2.10.1
* Node version: 8.9.2
* Operating System and version: Win10

**Comment by sverweij** [comment]
Thanks for raising this!

This needs to be fixed indeed. Probably using `path.possix` instead of
`path` for representations of path within dependency-cruiser code will
resolve the lion's share of this issue.

I'll pick it up in the coming day(s).

**Comment by sverweij** [comment]
hi @FaKeller 
- I've published `dependency-cruiser@2.11.0-beta-1`
  - it works consistently on win32 and posix platforms. 
  - in a few corner cases it doesn't yet on win32. I want to solve them before release 2.11.0, but it's not likely you'll bump into them
    - windows extended length paths don't resolve back to files on disk from the internal representation
    - neither do paths with non-ascii characters (`C:\\序列圖\\安全的交互.mscgen`)
    - the internal API doesn't handle basePaths correctly on win32. Only occurs if you access the API directly as the command line interface doesn't expose basePath as a parameter (yet)

Could you give it a spin and share your findings?

**Comment by FaKeller** [comment]
I just tried it and it works great! 👍👍👍  And as the rules work now I already spotted a bug 💃 Thank you for making this great tool!

**Comment by sverweij** [comment]
Good to hear it works well for you! (And thanks for the compliment 😊 ).

PR #29 is merged and is released on npmjs as version 2.11.0. Besides the cross platform support you've tested, the new release will handle corner cases mentioned above:
- windows extended length paths and non-ascii paths work
- basePaths passed to the internal API work 100% correct on win32 as well.
